test/py/ai_logging.py

```diff
@@ -0,0 +1,105 @@
+import argparse
+import os
+from time import sleep
+
+import numpy as np
+
+from dftracer.logger import dftracer
+from dftracer.logger.ai import ai
+
+
+class IOHandler:
+    @ai.data.item
+    def read(self, filename):
+        return np.load(filename)
+
+    def write(self, filename, a):
+        with open(filename, "wb") as f:
+            np.save(f, a)
+
+
+def get_args():
+    parser = argparse.ArgumentParser(
+        prog="DFTracer testing",
+        description="What the program does",
+        epilog="Text at the bottom of help",
+    )
+    parser.add_argument(
+        "--log_dir",
+        default="./pfw_logs",
+        type=str,
+        help="The log directory to save to the tracing",
+    )
+    parser.add_argument(
+        "--data_dir",
+        default="./data",
+        type=str,
+        help="The directory to save and load data",
+    )
+    parser.add_argument("--num_files", default=1, type=int, help="Number of files")
+    parser.add_argument(
+        "--niter", default=1, type=int, help="Number of iterations for the experiment"
+    )
+    parser.add_argument(
+        "--record_size",
+        default=1048576,
+        type=int,
+        help="size of the record to be written to the file",
+    )
+    args = parser.parse_args()
+    return args
+
+
+def data_gen(args, io: IOHandler, data):
+    for i in range(args.num_files):
+        io.write(f"{args.data_dir}/npz/{i}-of-{args.num_files}.npz", data)
+
+
+@ai.dataloader.fetch
+def read_data(args, io: IOHandler, epoch):
+    for i in range(args.num_files):
+        yield io.read(f"{args.data_dir}/npz/{i}-of-{args.num_files}.npz")
+
+
+@ai.device.transfer
+def transfer(data):
+    sleep(0.1)
+    return data
+
+
+@ai.compute.forward
+def forward(data):
+    sleep(0.1)
+    return 0.0
```

Comment:
If we have a function like @ai.compute if so lets add it here in the test.

```python
@ai.compute
def compute(step):
    forward()
    backward()
```

Reply:
I put it as `ai.compute.step` is this good? or you prefer `ai.compute` instead?